src/modules/MouseUtils/MouseJumpUI/Common/NativeMethods/User32/UI/Input/KeyboardAndMouse/User32.LPINPUT.cs

```diff
@@ -52,7 +52,7 @@ private static IntPtr ToPtr(INPUT[] values)
             var size = INPUT.Size;
             foreach (var value in values)
             {
-                Marshal.StructureToPtr(value, ptr, true);
+                Marshal.StructureToPtr(value, ptr, false);
```

Comment:
The memory is only allocated a few lines earlier so I figured there was actually nothing to delete and it should have been ```false``` all along.

Happy to revert it if preferred though.

Reply:
That's what you're doing in other spots you use this as well 🤔 So I assume it must be correct, we didn't have reports of leaks. Although, I wonder. It it was incorrect, why didn't the free cause a crash? It might be correct to clean all those uses instead, right?

Reply:
I can take a look at the other instances - I must have just spotted this one while making other changes.

Looking at the documentation it says ```false``` **is** the correct value for this scenario:

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.structuretoptr?view=net-8.0

> On the first call to the StructureToPtr method after a memory block has been allocated, fDeleteOld must be false, because there are no contents to clear.

I'll take a look at the other uses of ```StructureToPtr``` and fix any others that need it, if you want to hold off on merging this PR for now...

Reply:
That's the only place where you used true, I've checked. Good to merge, then. Thank you!

Reply:
awesome. thanks for the thorough review :-)

reading the small print in the documentation i *think* ```false``` might only cause a problem if the structures being marshalled have fields that contain object references, because the runtime keeps a reference count on thosee objects until the ```fDeleteOld``` flag is set to ```true``` or ```DestroyStructure``` is called

> If you use the StructureToPtr method to copy a different instance to the memory block at a later time, specify true for fDeleteOld to remove reference counts for reference types in the previous instance. Otherwise, the managed reference types and unmanaged copies are effectively leaked.

We (I) possibly got lucky because all the fields on the marshalled structure are (eventually) value types. That's just speculation though and it's definitely better to comply with the documentation :-)